components/backend/git/operations.go

```diff
@@ -222,6 +222,50 @@ func ParseGitHubURL(gitURL string) (owner, repo string, err error) {
 	return "", "", fmt.Errorf("not a GitHub URL")
 }
 
+// IsProtectedBranch checks if a branch name is a protected branch
+// Protected branches: main, master, develop
+func IsProtectedBranch(branchName string) bool {
+	protected := []string{"main", "master", "develop"}
+	normalized := strings.ToLower(strings.TrimSpace(branchName))
+	for _, p := range protected {
+		if normalized == p {
+			return true
+		}
+	}
+	return false
+}
+
+// GenerateBranchName creates a branch name from an RFE title
+// Converts to lowercase and replaces non-alphanumeric chars with hyphens
+// Returns the generated branch name and an error if it would create a protected branch
+func GenerateBranchName(title string) (string, error) {
```

Comment:
i feel like this should be done on the frontend, user gives branch name and then we validate it on this side